src/main/java/frc/team2412/robot/subsystem/IntakeSubsystem.java

```diff
@@ -1,13 +1,139 @@
 package frc.team2412.robot.subsystem;
 
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
 
 import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.I2C;
+import static frc.team2412.robot.subsystem.IntakeSubsystem.IntakeConstants.*;
 
 public class IntakeSubsystem extends SubsystemBase {
-    public IntakeSubsystem(WPI_TalonFX intakeMotor1, WPI_TalonFX intakeMotor2, DoubleSolenoid intakeSolenoid) {
 
+    // Constants
 
+    public static class IntakeConstants {
+        public static Alliance teamColor = DriverStation.getAlliance();
+
+        public static final double INTAKE_IN_SPEED = 0.5;
+        public static final double INTAKE_OUT_SPEED = -0.5;
+
+        public static final SupplyCurrentLimitConfiguration MAX_MOTOR_CURRENT = new SupplyCurrentLimitConfiguration(true, 40, 40, 500);
+        
+    }
+
+    // Enums
+
+    public enum IntakeMotorState {
+        IN, OUT, STOPPED;
+    }
+
+    public enum IntakeSolenoidState{
+        EXTENDED, RETRACTED;
+    }
+    
+    
+    public enum IntakeState{
+        EXTEND(DoubleSolenoid.Value.kForward),
+        RETRACT(DoubleSolenoid.Value.kReverse);
+
+        public DoubleSolenoid.Value value;
+
+        private IntakeState(DoubleSolenoid.Value value) {
+            this.value = value;
+        }
+    }
+    
+    
+    ///// IMPORTANT: Need ball amount variable and make method to stop taking in balls when at limit.
+
+    // Define Hardware
+
+    private final WPI_TalonFX motorOuterAxle;
+
+    private final WPI_TalonFX motorInnerAxle;
+
+    private final DoubleSolenoid solenoid;
+
+    ///private final ColorSensorV3 colorSensor = new colorSensorV3(i2cPort);
+
+    private IntakeMotorState intakeMotorState;
+
+    private IntakeSolenoidState intakeSolenoidState;
+
+    // CONSTRUCTOR!
+    public IntakeSubsystem(WPI_TalonFX motorOuterAxle, WPI_TalonFX motorInnerAxle, DoubleSolenoid intakeSolenoid) {
+        this.motorOuterAxle = motorOuterAxle;
+        this.motorInnerAxle = motorInnerAxle;
+        this.motorInnerAxle.setInverted(true);
+
+        this.motorInnerAxle.setNeutralMode(NeutralMode.Coast);
+        this.motorOuterAxle.setNeutralMode(NeutralMode.Coast);
+        
+
+        this.motorOuterAxle.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+        this.motorInnerAxle.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
```

Comment:
Ideally these can be set via configAllSettings() so all the other settings are reset to default, but that can be done in a subsequent PR